libusb/os/darwin_usb.c

```diff
@@ -1430,6 +1430,10 @@ static enum libusb_error get_endpoints (struct libusb_device_handle *dev_handle,
         return rc;
       }
 
+      if (iface >= config->bNumInterfaces) {
+        usbi_err (HANDLE_CTX (dev_handle), "interface out of range for device");
+        return LIBUSB_ERROR_NOT_FOUND;
```

Comment:
Shouldnt we use LIBUSB_ERROR_INVALID_PARAM like the checks in core.c?

Reply:
Yes, I can agree with INVALID_PARAM  when > max valid interface number (32), and NOT_FOUND when the device just doesn't have that value. Maybe libusb_detach_kernel_driver() should be changed?